packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -727,14 +989,287 @@ contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
     }
 
     /**
-     * @dev Internal function to handle the composed message logic - SIMPLIFIED
+     * @dev Decode fleet deposit compose message
+     * @param composeMessage The encoded compose message
+     * @return Decoded message data
+     */
+    function _decodeFleetDepositMessage(
+        bytes memory composeMessage
+    ) internal pure returns (FleetDepositMessageData memory) {
+        (
+            bytes32 messageType,
+            address fleetCommander,
+            address shareRecipient,
+            address asset,
+            uint256 amount,
+            uint256 sourceChainId,
+            bytes32 operationId,
+            address originalUser,
+            bytes memory referralCode
+        ) = abi.decode(
+                composeMessage,
+                (
+                    bytes32,
+                    address,
+                    address,
+                    address,
+                    uint256,
+                    uint256,
+                    bytes32,
+                    address,
+                    bytes
+                )
+            );
+
+        return
+            FleetDepositMessageData({
+                messageType: messageType,
+                fleetCommander: fleetCommander,
+                shareRecipient: shareRecipient,
+                asset: asset,
+                amount: amount,
+                sourceChainId: sourceChainId,
+                operationId: operationId,
+                originalUser: originalUser,
+                referralCode: referralCode
+            });
+    }
+
+    /**
+     * @dev Encode fleet deposit compose message
+     * @param data The message data to encode
+     * @return Encoded compose message
+     */
+    function _encodeFleetDepositMessage(
+        FleetDepositMessageData memory data
+    ) internal pure returns (bytes memory) {
+        return
+            abi.encode(
+                data.messageType,
+                data.fleetCommander,
+                data.shareRecipient,
+                data.asset,
+                data.amount,
+                data.sourceChainId,
+                data.operationId,
+                data.originalUser,
+                data.referralCode
+            );
+    }
+
+    /**
+     * @dev Internal function to handle the composed message logic - EXTENDED for Fleet Deposits
      */
     function _handleComposedMessage(
         address _from,
         uint256 amountLD,
         bytes memory composeMsg
     ) internal {
-        // Decode the compose message
+        // Get destination asset first
+        address destinationAsset = IStargateV2(_from).token();
```

Comment:
maybe `fleetAsset` ? since this is destination, but sounds like destination is the other chain :)